### PR TITLE
Python: Add scope entry definition nodes

### PR DESCRIPTION
**Pull request by yoff**
otherwise we confuse captured variables
in the single scope entry cfg node. Now
we have one for each defined variable.

### CODE REVIEW

**Review by hvitved — APPROVED** [review]

**Comment by yoff** [comment]
Evaluation came back as an optimization 🎉

File: python/ql/lib/semmle/python/dataflow/new/internal/DataFlowPrivate.qll [hunk 1]
```diff
@@ -484,8 +484,7 @@ predicate simpleLocalFlowStep(Node nodeFrom, Node nodeTo) {
  * or at runtime when callables in the module are called.
  */
 predicate simpleLocalFlowStepForTypetracking(Node nodeFrom, Node nodeTo) {
-  IncludePostUpdateFlow<PhaseDependentFlow<LocalFlow::localFlowStep/2>::step/2>::step(nodeFrom,
```
Comment (RasmusWL) [inline python/ql/lib/semmle/python/dataflow/new/internal/DataFlowPrivate.qll:484]: Can you give some more details? Ah, from looking over the implementation of LocalFlow::localFlowStep ([src](https://github.com/github/codeql/blob/faaa558ed90be76b41f0fa537f8391ee8f18dd3c/python/ql/lib/semmle/python/dataflow/new/internal/DataFlowPrivate.qll#L367-L378)) that already does the PhaseDependentFlow + IncludePostUpdateFlow stuff :+1:

**Comment by RasmusWL** [comment]
@yoff can you please highlight why the commit message for https://github.com/github/codeql/pull/15166/commits/07c88dc0be38177d2c9d13d79eb8daced4f2d27a says `also, it is slightly incorrect...`? What part is slightly incorrect now? :confused:

**Comment by yoff** [comment]
> @yoff can you please highlight why the commit message for [07c88dc](https://github.com/github/codeql/commit/07c88dc0be38177d2c9d13d79eb8daced4f2d27a) says `also, it is slightly incorrect...`? What part is slightly incorrect now? 😕

The PR description starts "remove unnecessary post-processing" and it is this post-processing that was slightly incorrect (in addition to being unnecessary). As it has been removed, there are no slightly incorrect parts left :-) (🤞)

**Comment by RasmusWL** [comment]
Ah, usually the old version is past-tense, so it would be more natural to say "also, it was slightly incorrect" -- at least that would have made me understand straight away :)

**Comment by yoff** [comment]
> Ah, usually the old version is past-tense, so it would be more natural to say "also, it was slightly incorrect" -- at least that would have made me understand straight away :)

Good point, I will keep that convention in mind going forward 👍

**Pull request merged**